src/main/java/gregtech/api/metatileentity/MetaPipeEntity.java

```diff
@@ -92,12 +96,16 @@ public MetaPipeEntity(int aID, String aBasicName, String aRegionalName, int aInv
         GT_LanguageManager.addStringLocalization("gt.blockmachines." + mName + ".name", aRegionalName);
         mInventory = new ItemStack[aInvSlotCount];
 
-        if (GT.isClientSide()) {
-            ItemStack tStack = new ItemStack(GregTech_API.sBlockMachines, 1, aID);
-            tStack.getItem().addInformation(tStack, null, new ArrayList<String>(), true);
+        if (aAddInfo && GT.isClientSide()) {
+            addInfo(aID);
         }
     }
 
+    protected final void addInfo(int aID) {
+    	ItemStack tStack = new ItemStack(GregTech_API.sBlockMachines, 1, aID);
```

Comment:
You should add server side protection here, otherwise if someplace calls this without the check it'll crash.

Reply:
Yep